include/nbl/video/IDescriptorPool.h

```diff
@@ -29,12 +33,156 @@ class NBL_API IDescriptorPool : public core::IReferenceCounted, public IBackendO
+            }
+
+            return (offset == invalidAddress) ? ~0u : offset;
+        }
+
+        void freeDescriptors(const uint32_t count, void* descriptors, const asset::E_DESCRIPTOR_TYPE type)
+        {
+            assert(m_flags & ECF_FREE_DESCRIPTOR_SET_BIT);
+            uint32_t allocatedAddr = static_cast<uint32_t>(reinterpret_cast<uint8_t*>(descriptors) - getDescriptorMemoryBaseAddress(type));
+            m_generalAllocators[type].free_addr(allocatedAddr, count*sizeof(core::smart_refctd_ptr<const asset::IDescriptor>));
+        }
+
+        inline uint8_t* getDescriptorMemoryBaseAddress(const asset::E_DESCRIPTOR_TYPE type) const
+        {
+            uint8_t* baseAddress;
+            switch (type)
+            {
+            case asset::EDT_COMBINED_IMAGE_SAMPLER:
+                baseAddress = reinterpret_cast<uint8_t*>(m_combinedImageSamplerStorage.get());
+                break;
+            case asset::EDT_STORAGE_IMAGE:
+                baseAddress = reinterpret_cast<uint8_t*>(m_storageImageStorage.get());
+                break;
+            case asset::EDT_UNIFORM_TEXEL_BUFFER:
+                baseAddress = reinterpret_cast<uint8_t*>(m_UTB_STBStorage.get());
+                break;
+            case asset::EDT_STORAGE_TEXEL_BUFFER:
+                baseAddress = reinterpret_cast<uint8_t*>(m_UTB_STBStorage.get()) + m_maxDescriptorCount[asset::EDT_UNIFORM_TEXEL_BUFFER] * sizeof(void*);
+                break;
+            case asset::EDT_UNIFORM_BUFFER:
+                baseAddress = reinterpret_cast<uint8_t*>(m_UBO_SSBOStorage.get());
+                break;
+            case asset::EDT_STORAGE_BUFFER:
+                baseAddress = reinterpret_cast<uint8_t*>(m_UBO_SSBOStorage.get()) + m_maxDescriptorCount[asset::EDT_UNIFORM_BUFFER] * sizeof(void*);
+                break;
+            case asset::EDT_UNIFORM_BUFFER_DYNAMIC:
+                baseAddress = reinterpret_cast<uint8_t*>(m_UBO_SSBOStorage.get()) + (m_maxDescriptorCount[asset::EDT_UNIFORM_BUFFER] + m_maxDescriptorCount[asset::EDT_STORAGE_BUFFER]) * sizeof(void*);
+                break;
+            case asset::EDT_STORAGE_BUFFER_DYNAMIC:
+                baseAddress = reinterpret_cast<uint8_t*>(m_UBO_SSBOStorage.get()) + (m_maxDescriptorCount[asset::EDT_UNIFORM_BUFFER] + m_maxDescriptorCount[asset::EDT_STORAGE_BUFFER] + m_maxDescriptorCount[asset::EDT_UNIFORM_BUFFER_DYNAMIC]) * sizeof(void*);
+                break;
+            case asset::EDT_INPUT_ATTACHMENT:
+                baseAddress = reinterpret_cast<uint8_t*>(m_storageImageStorage.get()) + m_maxDescriptorCount[asset::EDT_STORAGE_IMAGE] * sizeof(void*);
+                break;
+            case asset::EDT_ACCELERATION_STRUCTURE:
+                baseAddress = reinterpret_cast<uint8_t*>(m_accelerationStructureStorage.get());
+                break;
+            default:
+                assert(!"Invalid code path.");
+                return nullptr;
+            }
```

Comment:
let us not count in bytes, but index offsets, so this function nees to get removed

Reply:
no need for untyped pointers